tests/ui/pages/catalog/catalog-page.ts

```diff
@@ -48,8 +52,32 @@ export default class CatalogPage extends BasePage {
     return new InputField(this.page, SELECTORS.searchBar);
   }
 
-  async searchByText(text: string) {
+  async searchBy(text: string) {
     await this.searchBar.fill(text);
     await this.page.locator(SELECTORS.searchBar).press('Enter');
   }
+
+  async isAlertVisible(): Promise<boolean> {
+    await this.page.locator(SELECTORS.noMatchesFound).waitFor({ state: 'visible' });
+    return this.page.locator(SELECTORS.noMatchesFound).isVisible();
+  }
+
+  async isAlertHidden(): Promise<boolean> {
+    await this.page.locator(SELECTORS.noMatchesFound).waitFor({ state: 'hidden' });
+    return this.page.locator(SELECTORS.noMatchesFound).isHidden();
```

Comment:
convert the `this.page.locator(SELECTORS.noMatchesFound)` and `this.page.locator(SELECTORS.noMatchesFound)` into proper getters and reuse them